File: def_fucntion/ArbitaryArgumentList.py
def concat(*args, sep = '/'):
    return sep.join(args)

print(concat('earth', 'mars', 'venus', '谷军宇'))
print(concat('earth', 'mars', 'venus', '谷军宇', sep = ','))

def normal_concat(sep, *args):
    return sep.join(args)

print(normal_concat('-', 'earth', 'mars', 'venus', '谷军宇'))

File: def_fucntion/ArbitaryArgumentList.java
public class ArbitaryArgumentList {

    public static void main(String[] args) {
        System.out.println(concat("/", "earch", "mars", "venus", "谷军宇"));
        System.out.println(concat(",", "earch", "mars", "venus", "谷军宇"));
    }

    static String concat(String sep, String... args)  {
        java.util.StringJoiner sj = new java.util.StringJoiner(sep);
        for (String arg : args) {
            sj.add(arg);
        }

        return sj.toString();
    }
}
